subsys/bluetooth/host/classic/sco.c

```diff
@@ -312,7 +312,7 @@ uint8_t bt_esco_conn_req(struct bt_hci_evt_conn_request *evt)
 	}
 
 	sco_conn->role = BT_HCI_ROLE_PERIPHERAL;
```

Comment:
This I found a bit strange. I'm not familiar with Bluetooth Classic. Is it the case that it is the peripheral side which is the connecting side there?

Reply:
@jhedberg maybe you know?

Reply:
It could also be that for Bluetooth Classing the CONNECTING_STATE was used for both peripheral and central. We could consider creating a separate state for that

Reply:
> This I found a bit strange. I'm not familiar with Bluetooth Classic. Is it the case that it is the peripheral side which is the connecting side there?

Isn't the terminology "initiator" and "acceptor" for who initiates or accepts a connection? The resulting connection role is "peripheral" or "central" but that's something that's not strictly tied to the connection creation process due to the existence of the separate role switch procedure. That said, I think the SCO connections always derive their role form the corresponding ACL link, so instead of hardcoding the role here you should probably do the same.